src/metabase/lib/cache.cljc

```diff
@@ -17,12 +43,25 @@
      `WeakMap`s, using the input value `x` as the key and holding the cached result.
 
      For example, this works for caching by `:database-id` and then by legacy query object."
-     ([]                         #js {})
+     ([^js host]                 (if-let [cache (.-__mbcache host)]
+                                   cache
+                                   (set! (.-__mbcache host) #js {})))
      ([^js cache subkey x]       (when-let [inner-cache (gobject/get cache subkey)]
                                    (.get inner-cache x)))
      ([^js cache subkey x value] (let [inner-cache (gobject/setWithReturnValueIfNotSet cache subkey #(js/WeakMap.))]
                                    (.set inner-cache x value)))))
 
+(defn attach-query-cache
+  "Attaches the cache to a newly constructed query.
+
+  This uses metadata on CLJ and does nothing on CLJS."
+  [query]
+  #?(:cljs query
+     :clj  (vary-meta query (fn [mmeta]
+                              (cond-> mmeta
+                                (contains? mmeta :lib/__cache)
```

Comment:
What makes this true?

Reply:
I actually inverted this logic by accident while refactoring. It's fixed now.